### PR TITLE
[FEAT] Claim Achievement

### PR DESCRIPTION
**Pull request by HBandesh**
# Description

In this PR we have created an event to claim the achievement.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The feature was developed using TDD. Unit test were added to test the functionality under different scenarios. 

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/types/achievements.ts [hunk 1]
```diff
@@ -120,6 +132,8 @@ export const ACHIEVEMENTS: () => Record<AchievementName, Achievement> = () => ({
       return bakedCakes;
     },
     requirement: 10000,
```
Comment (hujikol) [inline src/features/game/types/achievements.ts:134]: is this really the requirement for "Bake 13 cakes" ? or is it should be 13?

Comment (adamhannigan) [reply src/features/game/types/achievements.ts:134]: Eagle eye 👁️ 🦅 !!!

You are right, we will need to fix this

**Review by adamhannigan — DISMISSED** [review]
LGTM 🚀

**Review by Rofrtd — APPROVED** [review]
Nice one mate. 
:shipit:

**Pull request merged**